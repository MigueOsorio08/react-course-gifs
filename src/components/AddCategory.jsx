import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState("");

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const onSubmitEvent = (e) => {
        e.preventDefault();

        if (inputValue.trim().length <= 1) return;

        setInputValue('')
        onNewCategory(inputValue.trim());
    }

    return (
        <form onSubmit={onSubmitEvent} aria-label="form">
            <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange} />
            <input className="search" type="button" value="Seach" onClick={onSubmitEvent}/>
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}