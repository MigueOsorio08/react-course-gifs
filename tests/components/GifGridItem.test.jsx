import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';


describe('Pruebas sobre el <GifGridItem/>', () => {

    const title = "one punch";
    const url = "https://localhost/algo.jpg";

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el URL y el alt indicado', () => {
        const { getByAltText } = render(<GifGridItem title={title} url={url} />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe mostrar el titulo en el componente', () => {
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
});