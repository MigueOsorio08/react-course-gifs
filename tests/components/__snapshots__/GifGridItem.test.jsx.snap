// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <GifGridItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch"
      src="https://localhost/algo.jpg"
    />
    <p>
      one punch
    </p>
  </div>
</div>
`;
